server/workflowarchive/archived_workflow_server.go

```diff
@@ -45,7 +45,7 @@ func (w *archivedWorkflowServer) ListArchivedWorkflows(ctx context.Context, req
 		listOptions = *req.ListOptions
 	}
 
-	options, err := sutils.BuildListOptions(listOptions, req.Namespace, req.NamePrefix, "")
+	options, err := sutils.BuildListOptions(listOptions, req.Namespace, req.NamePrefix, "", "", "")
```

Comment:
Indeed I missed that part for archived workflows, I'll change that and add tests.

For `minStartedAt` and `maxStartedAt` I didn't used those because I wasn't sure if they were still used somehow, and because they are not based on the same attribute so the query will be different. I can rename in the API if you tell me it's safe.

Reply:
I tried to update the code like this, but it seems that `createdAfter` and `finishedBefore` have no effect, not sure they are passed down to the API call. I spent few time trying to understand what's going on, do you have any idea ? I must miss something 

```golang
var workflows wfv1.Workflows
createdAfter := ""
finishedBefore := ""
if flags.createdSince != "" {
	createdTime, err := argotime.ParseSince(flags.createdSince)
	if err != nil {
		return nil, err
	}
	createdAfter = createdTime.Format(time.RFC3339)
}
if flags.finishedBefore != "" {
	finishedTime, err := argotime.ParseSince(flags.createdSince)
	if err != nil {
		return nil, err
	}
	finishedBefore = finishedTime.Format(time.RFC3339)
}

for {
	log.WithField("listOpts", listOpts).Debug()
	wfList, err := serviceClient.ListWorkflows(ctx, &workflowpkg.WorkflowListRequest{
		Namespace:      flags.namespace,
		ListOptions:    listOpts,
		Fields:         flags.displayFields(),
		CreatedAfter:   createdAfter,
		FinishedBefore: finishedBefore,
	})
	if err != nil {
		return nil, err
	}
	workflows = append(workflows, wfList.Items...)
	if wfList.Continue == "" {
		break
	}
	listOpts.Continue = wfList.Continue
}
```

Reply:
🚢 📦